### DISCUSSION TITLE
What is the difference between installation.created and installation_repositories.added

### DISCUSSION THREAD

**Discussion by SayakMukhopadhyay** [community/community#24379] [Apps, API and Webhooks]
<p>I am just starting out on developing my first GitHub app and I am not sure if I understand the use of the installation resource yet. My main concern now is that I am unable to understand the difference between the <code>installation</code> and the <code>installation_repositories</code> webhooks. When are they triggered and do I need to handle these events differently?</p>

**Comment by nethgato** [comment]
<p>Hey, hey <a class="mention" href="/sayakmukhopadhyay">@sayakmukhopadhyay</a>!</p>
<p>Been a while on this post, but I’m going back through the API category for unanswered topics and didn’t want to let this slip through. It’s entirely possible you’ve already found your answer, but if anyone else comes looking, our docs speak to the differences.</p>
<p>For <code>installation</code>:</p>
<p><a href="https://docs.github.com/en/developers/webhooks-and-events/webhook-events-and-payloads#installation" class="onebox" target="_blank">https://docs.github.com/en/developers/webhooks-and-events/webhook-events-and-payloads#installation</a></p>
<blockquote>
<p>Activity related to a GitHub App installation. The type of activity is specified in the  <code>action</code>  property of the payload object.</p>
</blockquote>
<p>…and note:</p>
<blockquote>
<p><strong>Note:</strong>  This event replaces an event that will be deprecated. When you subscribe to this event, you will also receive the deprecated event  <code>integration_installation</code>  until it’s permanently removed. If you’re writing or updating API client code, use the  <code>installation</code>  event. The event contain the same information as the deprecated event.</p>
</blockquote>
<p>Then, for <code>installation_repositories</code>:</p>
<p><a href="https://docs.github.com/en/developers/webhooks-and-events/webhook-events-and-payloads#installation_repositories" class="onebox" target="_blank">https://docs.github.com/en/developers/webhooks-and-events/webhook-events-and-payloads#installation_repositories</a></p>
<blockquote>
<p>Activity related to repositories being added to a GitHub App installation. The type of activity is specified in the  <code>action</code>  property of the payload object.</p>
</blockquote>
<p>…and note:</p>
<blockquote>
<p><strong>Note:</strong>  This event replaces an event that will be deprecated. When you subscribe to this event, you will also receive the deprecated event  <code>integration_installation_repositories</code>  until it’s permanently removed. If you’re writing or updating API client code, use the  <code>installation_repositories</code>  event. The event contain the same information as the deprecated event.</p>
</blockquote>
<hr>
<p>I’ve merely quoted our docs here, so please do let us know if there’s any remaining confusion! It’s possible our docs could do with some pruning and <a href="https://support.github.com/contact/feedback">feedback</a> is certainly welcome.</p>

**Comment by SayakMukhopadhyay** [comment]
<p>Thanks for jumping in. My main concern was knowing when exactly are these events triggered. Things like if one of the events can be triggered without the other Some examples would go a long way in understanding. Use cases and such.</p>

**Comment by nethgato** [comment, marked as answer]
<p>Hey <a class="mention" href="/sayakmukhopadhyay">@sayakmukhopadhyay</a> apologies; yes, you did ask about the <code>when</code>, not just the differences between them!</p>
<p>So <code>installation</code> is related to events when your app is installed on an Org, or Repository. Since an install can happen at the Org level for all, or specific repositories, you might not get a subsequent webhook triggered on a repository within an Org where all repositories are selected.</p>
<p>For <code>installation_repositories</code>, the webhook is there for those specific repositories that install your app. So say on the initial installation to an Org where they apply it to <code>all</code> repositories, you would only see the install event for the Org, but then you would see subsequent <code>installation_repositories</code> events, when new repositories are added to that Org, which then include your app as part of the creation process.</p>
<p>I really hope that helps clear up what the docs mean, but let us know if you have any more questions!</p>
